### ISSUE TITLE
Streamline cleaning functionality in `fourier_filter` and `vis_clean`

### ISSUE THREAD

**Issue by anguta** [HERA-Team/uvtools#70]
`fourier_filter` has lots of nearly duplicate code from `vis_clean` for doing cleaning. However, `vis_clean` does not support cleaning of arbitrary numbers areas not centered around delay=0. If we generalize `vis_clean` to support off center cleaning, we can just have `fourier_filter` call it and avoid lots of redundant code.

**Comment by jsdillon** [comment]
Wouldn't it make more sense to go the other way? Have `vis_clean` call `fourier_filter`?

**Comment by anguta** [comment]
> Wouldn't it make more sense to go the other way? Have `vis_clean` call `fourier_filter`?

Yes.

**Comment by anguta** [comment]
> > Wouldn't it make more sense to go the other way? Have `vis_clean` call `fourier_filter`?
> 
> Yes.

I agree with @jsdillon  here. Let's rename `high_pass_fourier_filter` to something like `vis_clean`, throw out the half-baked `dayenu` functionality,  and have it call `fourier_filter`

**Comment by jsdillon** [comment]
I'm fine with that as long as we maintain a `high_pass_fourier_filter` function (even as a thin wrapper) to maintain backwards compatibility.

**Comment by jsdillon** [comment]
Also, I would suggest not naming it visclean, since it's also used on calibration solutions. Maybe `CLEAN_filter` or `clean_filter`?

**Comment by anguta** [comment]
We've agreed to move `fitting_options` to `kwargs` in `fourier_filter`

**Comment by anguta** [comment]
![](https://files.slack.com/files-pri/T0KMWB8P3-F013ZKHHASZ/screen_shot_2020-05-20_at_5.33.03_pm.png)
